docs/upgrade-marshmallow.rst

```diff
@@ -0,0 +1,94 @@
+Marshmallow v3 compatibility
+============================
+
+Marshmallow
+-----------
+
+Marhsmallow is a package for serialization and deserialization of complex data
+structures from and to Python types. You can discover its usage within
+Invenio in the `documentation <https://invenio.readthedocs.io/en/latest/tutorials/understanding-data-models.html?highlight=marshmallow#define-a-marshmallow-schema>`_.
+
+
+Compatibility of v2 and v3
+--------------------------
+
+If you are using or you are about to use Marshmallow in your repository
+implementation, you should know that the package is about to be upgraded
+to version 3.x. There have been significant changes in the code which will
+influence already existing code and you should be aware of them.
+
+.. note::
+
+    The full guide is available in the Marshmallow `documentation <https://invenio.readthedocs.io/en/latest/tutorials/understanding-data-models.html?highlight=marshmallow#define-a-marshmallow-schema>`_.
+
+
+We are upgrading the Invenio modules to be compatible with Marshmallow v3.x.
+**We will provide a transition period in which modules will support both
+v2.3.x and v3.x implementations of Marshmallow**.
+Deprecated methods will be marked with warnings to provide you with
+time to adjust your code to the latest changes.
+
+
+There are two options you could follow if you are using
+schemas from Marshmallow 2:
+
+1. Stay with Marshmallow 2 for now by pinning the Marshmallow version
+in your instance.
+
+2. Upgrade to Marshmallow 3. Follow the guide below and the Marshmallow
+`documentation <https://invenio.readthedocs.io/en/latest/tutorials/understanding-data-models.html?highlight=marshmallow#define-a-marshmallow-schema>`_
+
+
+Guide
+-----
+
+After upgrading Marshmallow 3 in your dependencies you might find some of your
+code failing. Mostly it will be connected with schema validation and the `webargs <https://webargs.readthedocs.io/en/latest/quickstart.html>`_
+package. Webargs in version 5.4.0 is compatible with Marshmallow 3.x.
+
+The biggest change is the return type of ``Schema().dump()``,
+``Schema().load()``, ``Schema().dumps()`` and ``Schema().loads()``.
+They do not return the ``(data, errors)`` named tuple
+and the ``ValidationError`` is raised instead.
+In this case if you were using ``.errors`` attribute, it will no longer work
+- you need to implement your own handlers for the type
+``marshmallow.ValidationError``.
+Alternatively by doing ``Schema.validate()`` you can still get a list
+of the errors, an example is
+located in the `Marshmallow documentation <https://marshmallow.readthedocs.io/en/3.0/quickstart.html#schema-validate>`_.
+
+Other significant changes and solutions:
+
+- ``ValidationError`` - usually result of the schema becoming strict.
```

Comment:
I would pick something more along the lines of: 'Exception raised when marshmallow encounters an error during the validation, since the Exceptions are raised listed you can catch them and then return them as a dictionary to mimic the previous behaviour.'